chore(docs) Capacity planning (#1100)
questdb/questdb.io: docs/operations/advanced-capacity-planning.md

[Prometheus monitoring](/docs/third-party-tools/prometheus/) before taking one
of the following steps:

- Disabling the symbol cache. See
  [Usage of `symbols`](/docs/concept/symbol#usage-of-symbols) for server-wide
  and table-wide configuration options.
- Increasing the JVM heap size using the `-Xmx` argument.

### Symbol capacity

[Symbol capacity](/docs/concept/symbol#usage-of-symbols) should be the same or
slightly larger than the count of distinct symbol values.

Undersized symbol columns slow down query performance. Similarly, there is a
performance impact when symbol is not used for its designed way, most commonly
assigning `symbol` to columns with too many distinct values. It is crucial to
choose a suitable [data type](/docs/reference/sql/datatypes/) based on the
nature of the dataset.

### Index

Appropriate us of [indexes](/docs/concept/indexes/) provides faster read access
to a table. However, indexes have a noticeable cost in terms of disk space and
ingestion rate - we recommend starting with no indexes and adding them later,
only if they appear to improve query performance. Refer to
[Index trade-offs](/docs/concept/indexes#trade-offs) for more information.

> ## CPU configuration

Reviewer: The name of this section is weird as sub-sections include RAM and even storage. Should be something like `CPU and RAM configuration`

This section describes configuration strategies based on the forecast behavior
of the database.

> ### RAM size

Reviewer: I'd add one more section before this one: `CPU cores`. It should mention that we recommend at least 4 cores for basic workloads and 16 for advanced ones. Other than that, it should mention that the database tries to use all available CPU cores by default (with a link to `capacity-planning/#cpu-configuration` for those who want to understand that thread pool sizes and change them). Finally, we should mention that the throughput of read-only queries scales nicely in terms of the available cores (unless disk is the bottleneck, of course), so those who want better query performance, may want to use a machine with more cores.

Author: > It should mention that we recommend at least 4 cores for basic workloads and 16 for advanced ones.
I added the above to "capacity planning" as it seems to be more suitable.

Reviewer: Then maybe we should move the first sentence from RAM size section there as well? It makes no sense to have them in separate docs

We recommend having at least 8GB of RAM for basic workloads and 32GB for more
advanced ones.

For relatively small datasets, typically a few to a few dozen GB, if the need
for reads is high, performance can benefit from maximizing the use of the OS
page cache. Users may consider increasing available RAM to improve the speed of
read operations.

> ### Write amplification

Reviewer: To me, this section belongs to storage.

When ingesting out-of-order data, high disk write rate combined with high write
amplification may slow down the performance.

For data ingestion over InfluxDB Line Protocol (ILP), the first step to resolve
this start from adjusting the
[commit lag](/docs/guides/out-of-order-commit-lag/) value and the
`cairo.max.uncommitted.rows` value.

For data ingestion over PGWire, or as a further step for ILP ingestion, smaller
table [partitions](/docs/concept/partitions/) maybe reduce the write
amplification. This applies to tables with partition directories exceeding a few
hundred MBs on disk. For example, partition by day can be reduced to by hour,
partition by month to by day, and so on.

:::note

- For more information on commit lag, please refer to the
  [ILP commit strategy page](/docs/reference/api/ilp/tcp-receiver/#commit-strategy).
- In QuestDB the write amplification is calculated by the

>   [metrics](/docs/third-party-tools/prometheus#scraping-prometheus-metrics-from-questdb)

Reviewer: ```suggestion
  [metrics](/docs/third-party-tools/prometheus#scraping-prometheus-metrics-from-questdb): 
```

  `questdb_physically_written_rows_total` / `questdb_committed_rows_total`.
- Partitions are defined when a table is created. Refer to
  [CREATE TABLE](/docs/reference/sql/create-table/) for more information.

:::

### Memory page size configuration

For frequent out-of-order (O3) writes over high number of columns/tables, the
performance may be impacted by the size of the memory page being too big as this

> increases the demand for RAM.16M by default. The memory page,

Reviewer: ```suggestion
increases the demand for RAM. The memory page,
```

> `cairo.o3.column.memory.size`, is set to 16M by default. Decreasing the value in

Reviewer: ```suggestion
`cairo.o3.column.memory.size`, is set to 8M by default. This means that the table writer uses 16MB (2x8MB) RAM per each column when it receives O3 writes. Decreasing the value in
```

of data and filesystem considerations.

### Symbol caching

> <!--L3-->

Reviewer: That's probably L2

Reviewer 2: agree

- Increasing the JVM heap size using the `-Xmx` argument.

### Symbol capacity

> <!--L3-->

Reviewer: Same as above?

of the database.

### CPU cores

> <!--L2-->

Reviewer: This one sounds more like L3 to me

more cores will provide better query performance.

### RAM size

> <!--L2-->

Reviewer: Looks like L3 as well
